### PR TITLE
Add resource contention considerations

### PR DESCRIPTION
**Pull request by anssiko**
This is a first stab to port comments from @RafaelCintron @reillyeon in the TAG issue https://github.com/w3ctag/design-reviews/issues/933 to the specification to address feedback from the TAG. Please suggest changes to make sure key points are captured.

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/pull/765.html" title="Last updated on Oct 24, 2024, 1:12 PM UTC (2a441b0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/765/81a355a...2a441b0.html" title="Last updated on Oct 24, 2024, 1:12 PM UTC (2a441b0)">Diff</a>

### CODE REVIEW

**Review by reillyeon — APPROVED** [review]

**Comment by anssiko** [comment]
PTAL @RafaelCintron, I appreciate a review from you because this in part builds upon a discussion you had with the TAG.

File: index.bs [hunk 1]
```diff
@@ -609,6 +609,8 @@ This API is disabled by default in all cross-origin frames using the [[#permissi
 
 This API allows creation of an {{MLContext}} from a {{GPUDevice}} defined by WebGPU specification. See <a href="https://gpuweb.github.io/gpuweb/#security-considerations">WebGPU Security Considerations</a> for more information regarding security characteristics of this context.
 
+This API provides an abstraction across GPU, CPU and dedicated ML accelerator hardware. When using a GPU, <a href="https://www.w3.org/TR/webgpu/#security-dos">denial of service</a> considerations similar to WebGPU apply. When using a CPU or a dedicated ML accelerator, the types of potential resource contention are different and mitigations will be implementation- and configuration-dependent. Implementations should use whatever mechanisms available from the platform to prevent sites from using an unfair amount of system resources. These compute units are shared resources and the use of any compute API will affect overall performance on a fully-loaded system.
```
Comment (fdwr) [inline index.bs:612]: The trailing dash after "implementation" looks weird. Propose: `implementation and configuration dependent"

Comment (fdwr) [inline index.bs:612]: > Implementations should use whatever mechanisms [are] available from the platform

Missing word.

Comment (fdwr) [inline index.bs:612]: >  are shared resources[,] and the use of

Comma after coordinating conjunction of independent clauses.

**Review by fdwr — APPROVED** [review]
Approved with grammar suggestions. @RafaelCintron may have thoughts too.

**Comment by anssiko** [comment]
@fdwr thank you for the review and grammar suggestions.

**Comment by anssiko** [comment]
I'll merge this PR now to focus our open PR queue. I'll also inform the TAG that we've addressed their feedback.

**Review by huningxin — COMMENTED** [review]
LGTM!

**Pull request merged**